aspnet/fundamentals/owin.rst

```diff
@@ -1,8 +1,386 @@
 
-|stub-icon| OWIN
-================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET 5 supports OWIN, the Open Web Interface for .NET, which allows web applications to be decoupled from web servers. In addition, OWIN defines a standard way for middleware to be used in a pipeline to handle individual requests and associated responses. ASP.NET 5 applications and middleware can interoperate with OWIN-based applications, servers, and middleware.
 
-.. _issue: https://github.com/aspnet/Docs/issues/71
\ No newline at end of file
+In this article:
+	- `Running OWIN middleware in the ASP.NET pipeline`_
+	- `Using ASP.NET Hosting on an OWIN-based server`_
+	- `Run ASP.NET 5 on an OWIN-based server and use its WebSockets support`_
+	- `OWIN keys`_
+
+`Browse or download samples on GitHub <https://github.com/aspnet/Docs/tree/master/docs/fundamentals/owin/sample>`_.
+
+Running OWIN middleware in the ASP.NET pipeline
+-----------------------------------------------
+
+ASP.NET 5's OWIN support is deployed as part of the `HttpAbstractions package <https://github.com/aspnet/HttpAbstractions>`_, in the Microsoft.AspNet.Owin project. You can import OWIN support into your project by adding this package as a dependency in your ``project.json`` file, as shown here:
+
+.. literalinclude:: owin/sample/src/OwinSample/project.json
+	:language: javascript
+	:lines: 5-10
+	:emphasize-lines: 3
+
+OWIN middleware conform to the `OWIN specification <http://owin.org/spec/spec/owin-1.0.0.html>`_, which defines a ``Properties IDictionary<string, object>`` interface that must be used, and also requires certain headers be set (such as ``owin.ResponseBody``). We can construct a very simple example of middleware that follows the OWIN specification to display "Hello World", as shown here:
+
+.. literalinclude:: owin/sample/src/OwinSample/Startup.cs
+	:language: c#
+	:lines: 26-39
+	:dedent: 8
+
+In the above example, notice that the method returns a ``Task`` and accepts an ``IDictionary<string, object>`` as required by OWIN. Within the method, this parameter is used to specify the ``owin.ResponseBody`` and ``owin.ResponseHeaders`` objects within the environment dictionary. Once the headers are set appropriately for the content being returned, a task representing the asynchronous write to the response stream is returned.
+
+Adding OWIN middleware to the ASP.NET pipeline is most easily done using the ``UseOwin`` extension method. Given the ``OwinHello`` method shown above, adding it to the pipeline is a simple matter:
+
+.. literalinclude:: owin/sample/src/OwinSample/Startup.cs
+	:language: c#
+	:lines: 18-24
+	:dedent: 8
+
+You can of course configure other actions to take place within the OWIN pipeline. Whether directly from within the call to ``UseOwin`` as shown below, or through multiple calls to ``UseOwin``. Just remember that response headers should only be modified prior to the first write to the response stream, so configure your pipeline accordingly.
```

Comment:
Multiple calls to UseOwin is discouraged for performance reasons, OWIN components will operate best if grouped together.